File: api/TodoApi/Domain/Entities/TodoItem.cs
using System;

namespace TodoApi.Domain.Entities
{
    public class TodoItem
    {
        public Guid Id { get; private set; }

        public string Name { get; private set; }

        public bool Completed { get; private set; }

        private TodoItem(string name)
        {
            Id = Guid.NewGuid();
            Name = name;
        }

        private TodoItem()
        {
        }

        public static TodoItem Create(string name)
             => new TodoItem(name);

        public void Toogle()
        {
            Completed = !Completed;
        }
    }
}

File: Web/ClientApp/src/model/todoItem.ts
import { Guid } from 'guid-typescript'

export default class TodoItem {
    Id: Guid;
    Name: string;
    Completed: boolean;

    constructor(id: Guid, name: string) {
        this.Id = id;
        this.Name = name;
        this.Completed = false;
    }

    toogle = () => {
        this.Completed = !this.Completed;
    } 
}